internal/provider/resource_network.go

```diff
@@ -316,9 +317,9 @@ func resourceNetworkGetResourceData(d *schema.ResourceData) (*unifi.Network, err
 	}, nil
 }
 
-func resourceNetworkSetResourceData(resp *unifi.Network, d *schema.ResourceData, site string) error {
+func resourceNetworkSetResourceData(resp *unifi.Network, d *schema.ResourceData, site string) diag.Diagnostics {
 	wanType := ""
-	wanDNS := []string{}
+	var wanDNS []string
```

Comment:
Not sure if the SDK/Unifi handle nulls differently than empty, but would rather leave this alone for this PR.
```suggestion
	wanDNS := []string{}
```